src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -227,6 +238,834 @@ class OMPSerialDirective(OMPRegionDirective, metaclass=abc.ABCMeta):
+        :raises UnresolvedDependencyError: If ref is a BinaryOperation and \
+                                           neither child of ref is a Literal \
+                                           or BinaryOperation.
+        :raises UnresolvedDependencyError: If there is a dependency between \
+                                           ref (a BinaryOperation) and a \
+                                           previously set constant.
+        :raises UnresolvedDependencyError: If there is a dependency between \
+                                           ref and a Loop variable that is \
+                                           not an ancestor of task.
+        :raises UnresolvedDependencyError: If preceding_nodes contains a \
+                                           dependent loop with a non-Literal \
+                                           step.
+
+        :returns: a list of the dependency values for the input ref.
+        :rtype: List[Union[:py:class:`psyclone.psyir.nodes.Literal`, \
+                :py:class:`psyclone.psyir.nodes.BinaryOperation`]]
+        '''
+        if isinstance(ref, Reference):
+            symbol = ref.symbol
+        else:
+            # Get the symbol out of the Binop, and store some other
+            # important information. We store the step value of the
+            # ancestor loop (which will be the value of the Literal, or
+            # the one of the Literals if an operand is a BinaryOperation).
+            # In the case that one of the operands is a BinaryOperation,
+            # we also store a "num_entries" value, which is based upon the
+            # multiplier of the step value. This is how we can handle
+            # cases such as array(i+57) if the step value is 32, the
+            # dependencies stored would be array(i+32) and array(i+64).
+            if isinstance(ref.children[0], Literal):
+                if ref.operator == BinaryOperation.Operator.ADD:
+                    # Have Literal + Reference. Store the symbol of the
+                    # Reference and the integer value of the Literal.
+                    symbol = ref.children[1].symbol
+                    binop_val = int(ref.children[0].value)
+                    num_entries = 2
+                else:
+                    raise UnresolvedDependencyError(
+                            "Found a dependency index that is "
+                            "a BinaryOperation where the "
+                            "format is Literal OP Reference "
+                            "with a non-ADD operand "
+                            "which is not supported.")
+            elif isinstance(ref.children[1], Literal):
+                # Have Reference OP Literal. Store the symbol of the
+                # Reference, and the integer value of the Literal. If the
+                # operator is negative, then we store the value negated.
+                if ref.operator == BinaryOperation.Operator.ADD:
```

Comment:
This might be slightly nicer as:

    if ref.operator in [BinaryOperation.Operator.ADD, BinaryOperation.Operator.SUB]:
        symbol = ref.children[0].symbol
        num_entries = 2
        binop_val = int(ref.children[1].value)
        if ref.operator == BinaryOperation.Operator.SUB:
            binop_val = -binop_val

Reply:
Could you also do a similar thing for L430-480?

Reply:
I did this, the second one has slightly different value for `num_entries` in the negative case too, I've added my guess as to why that was but I'm not totally certain...